x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MlDailyMaintenanceService.java

```diff
@@ -151,6 +151,10 @@ private void triggerTasks() {
                 LOGGER.warn("skipping scheduled [ML] maintenance tasks because upgrade mode is enabled");
                 return;
             }
+            if (MlMetadata.getMlMetadata(clusterService.state()).isResetMode()) {
+                LOGGER.warn("skipping scheduled [ML] maintenance tasks because reset mode is enabled");
```

Comment:
++ good catch

This means we have to be careful not to leave reset mode enabled and should try to ensure `cleanUpFeature` does not leave it set.